core/templates/dev/head/collection_editor/editor_tab/CollectionLinearizerServiceSpec.js

```diff
@@ -0,0 +1,468 @@
+          newCollectionNode.getExplorationSummaryObject());
+      newCollectionNode = collection.getCollectionNodeByExplorationId(
+        'exp_id3');
+      expect(newCollectionNode.getPrerequisiteSkillList().getSkills()).toEqual(
+          ['exp title2']);
+      expect(newCollectionNode.getAcquiredSkillList().getSkills()).toEqual(
+          ['exp title3']);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([
+                collection.getCollectionNodeByExplorationId('exp_id0'),
+                collection.getCollectionNodeByExplorationId('exp_id1'),
+                collection.getCollectionNodeByExplorationId('exp_id2'),
+                collection.getCollectionNodeByExplorationId('exp_id3')]);
+    });
+  });
+
+  describe('shiftNodeLeft()', function() {
+    it('should handle shifting a node in a single node collection',
+        function() {
+      var collection = CollectionObjectFactory.createEmptyCollection();
+      firstCollectionNode.getAcquiredSkillList().setSkills(['exp title0']);
+      collection.addCollectionNode(firstCollectionNode);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode]);
+      expect(CollectionLinearizerService.shiftNodeLeft(
+          collection, 'exp_id0')).toBe(true);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode]);
+    });
+
+    it('should not shift a non-existent node', function() {
+      var collection = createLinearCollection();
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, secondCollectionNode,
+                  thirdCollectionNode]);
+      expect(CollectionLinearizerService.shiftNodeLeft(
+          collection, 'non_existent')).toBe(false);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, secondCollectionNode,
+                  thirdCollectionNode]);
+    });
+
+    it('should handle shifting the first node', function() {
+      var collection = createLinearCollection();
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, secondCollectionNode,
+                  thirdCollectionNode]);
+      expect(CollectionLinearizerService.shiftNodeLeft(
+          collection, 'exp_id0')).toBe(true);
+      expect(
+          CollectionLinearizerService.getCollectionNodesInPlayableOrder(
+              collection)).toEqual([firstCollectionNode, secondCollectionNode,
+                  thirdCollectionNode]);
+    });
+
+    iit('should handle shifting the last node', function() {
```

Comment:
This should be ``it()``. The other tests aren't going to be able to run, currently.